pkg/controller/periodic/periodic.go

```diff
@@ -67,10 +67,14 @@ func (c *Controller) Run(stopCh <-chan struct{}, initialDelay time.Duration) {
 		case <-stopCh:
 			return
 		case <-time.After(initialDelay):
-			c.Gather()
+			if c.configurator.Config().Report {
```

Comment:
Can't we move this if statement inside the `Gather()`? It will avoid some code duplication.